src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocFile.java

```diff
@@ -216,6 +239,32 @@ public void copyResource(DocPath resource, boolean overwrite, boolean replaceNew
         }
     }
 
+    private static final Pattern replacePtn = Pattern.compile("##REPLACE:(?<key>[A-Za-z0-9._]+)##");
+
+    private String localize(String line, Resources resources) {
+        Matcher m = replacePtn.matcher(line);
+        StringBuilder sb = null;
+        int start = 0;
+        while (m.find()) {
+            if (sb == null) {
+                sb = new StringBuilder();
+            }
+            sb.append(line, start, m.start());
+            try {
+                sb.append(resources.getText(m.group("key")));
+            } catch (MissingResourceException e) {
+                sb.append(m.group());
```

Comment:
Shouldn't we propagate an error here, or issue a warning? If a key is missing localized properties the value from default properties should be used, so this should never happen, right?

Reply:
I see the added check for "##REPLACE:" in TestSearch.java will catch that case, so I guess it's ok.